auth/src/main/java/com/firebase/ui/auth/util/ui/FlowHolder.java

```diff
@@ -0,0 +1,75 @@
+package com.firebase.ui.auth.util.ui;
+
+import android.app.Application;
+import android.app.PendingIntent;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.content.Intent;
+import android.support.annotation.RestrictTo;
+import android.util.Pair;
+
+import com.firebase.ui.auth.data.model.ActivityResult;
+import com.firebase.ui.auth.data.model.FlowParameters;
+import com.firebase.ui.auth.util.data.AutoClearSingleLiveEvent;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class FlowHolder extends ViewModelBase<FlowParameters> {
+    private static MutableLiveData<ActivityResult> ACTIVITY_RESULT_LISTENER =
+            new AutoClearSingleLiveEvent<>(new Runnable() {
+                @Override
+                public void run() {
+                    ACTIVITY_RESULT_LISTENER = new AutoClearSingleLiveEvent<>(this);
+                }
+            });
+    private static MutableLiveData<Pair<Intent, Integer>> INTENT_STARTER =
+            new AutoClearSingleLiveEvent<>(new Runnable() {
+                @Override
+                public void run() {
+                    INTENT_STARTER = new AutoClearSingleLiveEvent<>(this);
+                }
+            });
+    private static MutableLiveData<Pair<PendingIntent, Integer>> PENDING_INTENT_STARTER =
+            new AutoClearSingleLiveEvent<>(new Runnable() {
+                @Override
+                public void run() {
+                    PENDING_INTENT_STARTER = new AutoClearSingleLiveEvent<>(this);
+                }
+            });
```

Comment:
Here's where you get to admire the grossness of `AutoClearSingleLiveEvent`... oh well.